winfetch.ps1

```diff
@@ -671,29 +709,52 @@ function info_disk {
         }
     }
 
-    $disks = Get-CimInstance -ClassName Win32_LogicalDisk -Property Size,FreeSpace -CimSession $cimSession
-
-    foreach ($diskInfo in $disks) {
-        foreach ($diskLetter in $showDisks) {
-            if ($diskInfo.DeviceID -eq $diskLetter -or $diskLetter -eq "*") {
-                $total = $diskInfo.Size
-                $used = $total - $diskInfo.FreeSpace
-		if ($total -gt 0) {
-		    $usage = [math]::floor(($used / $total * 100))
-		    [void]$lines.Add(@{
-		        title   = "Disk ($($diskInfo.DeviceID))"
-		        content = get_level_info "" $diskstyle $usage "$(to_units $used) / $(to_units $total)"
-		    })
-		}
-                break
+    # Convert System.String[] to System.Object[]
+    $rawDiskLetters = [WinAPI.DiskMethods]::GetLogicalDriveStringsW()
+    $allDiskLetters = @()
+    foreach ($entry in $rawDiskLetters) {
+        if ($entry -ne ":" -and $entry -ne "\" -and $entry + ":" -ne ":") {
+            $allDiskLetters += $entry + ":"
+        }
+    }
+
+    # Verification stage
+    $diskLetters = @()
+    foreach ($diskLetter in $allDiskLetters) {
+        foreach ($showDiskLetter in $showDisks) {
+            if ($diskLetter -eq $showDiskLetter -or $showDiskLetter -eq "*") {
+                $diskLetters += $diskLetter
             }
         }
     }
 
+    foreach ($diskLetter in $diskLetters) {
+        $lpFreeBytesAvailable = 0
+        $lpTotalNumberOfBytes = 0
+        $lpTotalNumberOfFreeBytes = 0
+        $success = [WinAPI.DiskMethods]::GetDiskFreeSpaceEx($diskLetter, [ref] $lpFreeBytesAvailable, [ref] $lpTotalNumberOfBytes, [ref] $lpTotalNumberOfFreeBytes)
+        $total = $lpTotalNumberOfBytes
+        $used = $total - $lpTotalNumberOfFreeBytes
+
+        if (-not $success) {
+            [void]$lines.Add(@{
+                title   = "Disk ($diskLetter)"
+                content = "(failed to get disk usage)"
+            })
+        }
+        
+        if ($total -gt 0) {
+            $usage = [math]::floor(($used / $total * 100))
+            [void]$lines.Add(@{
+                title   = "Disk ($diskLetter)"
+                content = get_level_info "" $diskstyle $usage "$(to_units $used) / $(to_units $total)"
+            })
+        }
+    }
+
     return $lines
 }
 
-
```

Comment:
Let's keep this line, we have a gap of 2 newlines between each top level function.